chia/wallet/util/address_type.py

```diff
@@ -0,0 +1,55 @@
+from enum import Enum
+from typing import Any, Dict, Optional, Set
+
+from chia.util.bech32m import bech32_decode, convertbits
+from chia.util.config import selected_network_address_prefix
+
+
+class AddressType(Enum):
+    XCH = "xch"
+    NFT = "nft"
+    DID = "did:chia:"
+
+    def hrp(self, config: Optional[Dict[str, Any]] = None) -> str:
```

Comment:
While it may be worthwhile at the moment since we don't have `tnft` etc, making the config optional leaves open errors where even with an xch type you just forget to pass the config and so never get txch.  You can write a bit more code or have to be really super careful any time you touch a bit less code.

I won't require a change here to approve, but be really careful...

Reply:
Agreed that it makes it too easy to make a mistake. The config is now required here and in the validation functions.